### PR TITLE
Magento Catalog - fix custom option type text price conversion for mu…

### PR DESCRIPTION
**Pull request by oleksii-lisovyi**
### Description

Previously I already fixed similar issue in the PR https://github.com/magento/magento2/pull/18225, but the fix doesn't cover all the custom option types, e.g. type "Field" still has that issue with price conversion. 

_It's reproduced on the following Magento setup with multi-currency configuration:_
1. There are 2 websites configured: US and CA (or another one).
2. Base Currency is configured in the next way: default level - USD; CA - CAD.
3. And if you try to update product with custom options type "Field" (or another without option values) with non zero price on CA website level.

### Fixed Issues (if relevant)

No related issues found.

### Manual testing scenarios

**Steps to reproduce**
1. Create 2 websites on Magento: US website and CA website.
2. Set up Base Currency to USD (or any other) on default scope.
3. Set up Base Currency to CAD on CA website scope.
4. Create product with custom option type "Field" (or another without option values) and set up non zero price for option.
5. Save the product.
6. Change configuration scope on CA website.
7. Save product.

**Expected result**
Custom option price is not changed, because no actual changes were done.

**Actual result**
Custom option price is updated according to USD/CAD currency rate.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @oleksii-lisovyi. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.2-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-4767](https://jira.corp.magento.com/browse/ENGCOM-4767) has been created to process this Pull Request

**Comment by ihor-sviziev** [comment]
Hi @oleksii-lisovyi, in order to focus on the 2.3 release line we are accepting PRs to 2.3-develop first. Could you please create PR with the same changes to 2.3-develop and we will process current PR once it will be merged.

As for now I'm putting this PR on hold before PR to 2.3-develop branch will be merged.

Thank you!

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-4767](https://jira.corp.magento.com/browse/ENGCOM-4767) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Pull request merged**
